crates/storage/provider/src/post_state/mod.rs

```diff
@@ -242,38 +242,55 @@ impl PostState {
             .root()
     }
 
-    // todo: note overwrite behavior, i.e. changes in `other` take precedent
     /// Extend this [PostState] with the changes in another [PostState].
     pub fn extend(&mut self, mut other: PostState) {
-        // Update plain state
-        self.accounts.extend(other.accounts);
-        for (address, their_storage) in other.storage {
-            let our_storage = self.storage.entry(address).or_default();
-            if their_storage.wiped() {
-                our_storage.times_wiped += their_storage.times_wiped;
-                our_storage.storage.clear();
+        // Insert storage change sets
+        for (block_number, storage_changes) in std::mem::take(&mut other.storage_changes).inner {
+            for (address, their_storage_transition) in storage_changes {
+                let our_storage = self.storage.entry(address).or_default();
+                let (wipe, storage) = if their_storage_transition.wipe.is_wiped() {
+                    // Check existing storage change.
+                    match self.storage_changes.get(&block_number).and_then(|ch| ch.get(&address)) {
+                        Some(change) if change.wipe.is_wiped() => (), // already counted
+                        _ => {
+                            our_storage.times_wiped += 1;
```

Comment:
Not related to this, but I was looking in the code to see if there is `times_wiped-1`I am not sure if needed but was curious.

Reply:
that's a good point. i think this is missing from `split_at` function for `non_reverted_state`